modules/extensions-support/src/main/java/org/mule/module/extension/internal/introspection/ParameterDescriptor.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.introspection;
+
+import org.mule.extension.introspection.DataType;
+
+/**
+ * Intermediate representation of a parameter used to decouple
+ * an interchangeable introspection mechanism from the extension's API
+ * model
+ *
+ * @since 3.7.0
+ */
+final class ParameterDescriptor
```

Comment:
Can but would add no value. This is just a barrier class with a extremely short lifespan... No instance of this class will still be referenced by the time the MuleContext receives the start() call.

Making it immutable would only complicate the code